packages/xo-server/src/api/pci.mjs

```diff
@@ -0,0 +1,43 @@
+import { asyncEach } from '@vates/async-each'
+import { defer } from 'golike-defer'
+
+export const hide = defer(async function ($defer, { pcis, hide }) {
+  await this.checkPermissions(pcis.map(id => [id, 'administrate']))
+
+  const _pcis = pcis.map(id => this.getObject(id, 'PCI'))
+  if (!_pcis.every(pci => pci.$host === _pcis[0].$host)) {
+    // For the changes to take effect, the host must reboot.
+    // By only allowing PCIs within the same host, this avoids having to manage the restart of multiple hosts and its priority.
+    throw new Error('All PCIs must be in the same host')
+  }
+  const xapi = this.getXapi(_pcis[0])
+  const getMethod = hide => `PCI.${hide ? 'disable' : 'enable'}_dom0_access`
+
+  await asyncEach(
+    _pcis,
+    async pci => {
+      await xapi.call(getMethod(hide), pci._xapiRef)
+      $defer.onFailure(() => xapi.call(getMethod(!hide), pci._xapiRef))
```

Comment:
if the host fail to boot , it will go in a infinite loop (for example a broken patch that was waiting for reboot ) Since it's external to this method, I think a comment is enough, or  a third param to the method undoOnFailure (default true)